src/phpDocumentor/Compiler/Linker/Linker.php

```diff
@@ -267,17 +276,17 @@ protected function isDescriptorContainer($item): bool
     /**
      * Replaces pseudo-types, such as `self`, into a normalized version based on the last container that was
      * encountered.
-     *
-     * @param DescriptorAbstract|null $container
      */
-    protected function replacePseudoTypes(string $fqsen, $container): string
+    protected function replacePseudoTypes(string $fqsen, ?DescriptorAbstract $container) : string
     {
         $pseudoTypes = ['self', '$this'];
         foreach ($pseudoTypes as $pseudoType) {
-            if ((strpos($fqsen, $pseudoType . '::') === 0 || $fqsen === $pseudoType) && $container) {
-                $fqsen = $container->getFullyQualifiedStructuralElementName()
-                    . substr($fqsen, strlen($pseudoType));
+            if ((strpos($fqsen, $pseudoType . '::') !== 0 && $fqsen !== $pseudoType) || !$container) {
```

Comment:
If we remove the call with a `null` value for `$container` we could remove the or here.

Reply:
That requirement for a null value comes from deeper in the Linker; I do not dare change that right now :D I have added a todo to revisit this

Reply:
To be fair, there are quite a few places where stuff is nullable that would be nice if it were not. But I prefer not to fix that at once but instead go over the codebase as soon as we have passing stylechecks and improve those locations one by one. Same goes for mixed types that we have in places